packages/flutter_tools/lib/src/build_system/targets/dart_plugin_registraint.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:flutter_tools/src/dart/package_map.dart';
+import 'package:meta/meta.dart';
+import 'package:package_config/package_config.dart';
+
+import '../../base/file_system.dart';
+import '../../plugins.dart';
+import '../../project.dart';
+import '../build_system.dart';
+
+class DartPluginRegistrantTarget extends Target {
+
+  DartPluginRegistrantTarget({@required this.mainPath});
```

Comment:
This should be read from the environment configuration - see the KernelSnapshot target for an example, think it is called target or something like that